### ISSUE TITLE
Two feature requests for rowing integration

### ISSUE THREAD

**Issue by JaapvanEkris** [markw65/fit-file-writer#5]
First of all, thank you for your great work. I am using your NPM for adding fit-files export to OpenRowingMonitor (I'm the package maintainer for that), it is a breeze to create files this way.

I'm missing a field which is used in rowing machines (in the "record" message): the stroke_distance. I think it has ID 87, and seems to be in centimeters (values typically range between 900 cm and 1200 cm. When using cycle length it doesn't get picked up, and when I record on a Garmin, it is stored in the 87 field.

An additional request is to be able to reset the recorded file and start again. I know I can clear the entire variable, but a nice reset function would be cleaner I think.

Thanks again for the great work, and I hope you can help me.

**Comment by markw65** [comment]
Can you point me to the `stroke_distance` documentation? I'm not seeing it anywhere in the fit sdk.

You mention `cycle_length` - I'm not sure how that's connected to `stroke_distance`, but I do see a field named `cycle_length16` with id 87. It appears to be identical to cycle_length, except it's a `uint16` rather than a `uint8`. If the values are typically 900 to 1200cm that would make sense...

Is that what you're looking for?

**Comment by JaapvanEkris** [comment]
> You mention `cycle_length` - I'm not sure how that's connected to `stroke_distance`, but I do see a field named `cycle_length16` with id 87. It appears to be identical to cycle_length, except it's a `uint16` rather than a `uint8`. If the values are typically 900 to 1200cm that would make sense...
> 
> Is that what you're looking for?

Yep,

That is the one. One of my fellow developers on OpenRowingMonitor discovered the field two years ago, see his bugreport: https://forums.garmin.com/developer/fit-sdk/f/discussion/323579/missing-stroke_distance-field-on-the-record-message

He confirmed it is indeed called 'cycle_length16', but the averages are stored in 'avg_stroke_distance' on the lap and session level, just to maximise confusion....

It is a uint16, data is recorded in centimeters but its unit is in meters.

When I look at the file captured by a Garmin Epix of an indoor rowing session, we see it uses `cycle_length16` to record a rowing metric called 'distance per stroke', which is reported as a native Garmin field for indoor rowing. According to my fellow developer, it is poorly documented in the FIT documentation (see [our issue report here](https://github.com/JaapvanEkris/openrowingmonitor/issues/59))

**Comment by markw65** [comment]
[test1.fit.zip](https://github.com/user-attachments/files/17691881/test1.fit.zip)

Can you confirm this is what you're expecting? The field is defined with units of `m` with a scaling factor of 100. So the raw values are in `cm`. I modified my test case to include `cycle_length16` fields in the range 9 to 12m (so 900 to 1200cm). The resulting fit file can be decoded at https://www.fitfileviewer.com, and shows the values I was trying to write. I just want to make sure I've got the scaling correct.

**Comment by JaapvanEkris** [comment]
> Can you confirm this is what you're expecting? The field is defined with units of `m` with a scaling factor of 100. So the raw values are in `cm`. I modified my test case to include `cycle_length16` fields in the range 9 to 12m (so 900 to 1200cm). The resulting fit file can be decoded at https://www.fitfileviewer.com, and shows the values I was trying to write. I just want to make sure I've got the scaling correct.

This is indeed how it works! For your info, see
[Epix_5K_Indoor_Rowing.zip](https://github.com/user-attachments/files/17692750/Epix_5K_Indoor_Rowing.zip)

Which shows the exact same fields :)

**Comment by markw65** [comment]
v0.1.3 should fix the cycle_length16 issue.

I'm not sure I want to add a `reset()` method. What's the advantage vs throwing the old one away and creating a new one?

**Comment by JaapvanEkris** [comment]
> v0.1.3 should fix the cycle_length16 issue.

Thanks, I'll test tomorrow! That will make a lot of indoor rowers happy!

> I'm not sure I want to add a `reset()` method. What's the advantage vs throwing the old one away and creating a new one?

Yeah, I solved this differently now, by limiting the object scope to our file generation function. We noticed the Javascript garbage collector has some issues, which sometimes results in a memory leak. So we try to clear memory as explicit as we can (solved a lot of our memory issues in our own code). But our current approach works well enough
